### PR TITLE
7 create sqlalchemy connection for backend and 8 setup pytest for backend

### LINKED ISSUE #8: Setup pytest for backend
Setup backend testing using pytest.

Replace normal naming standards with the following pytest.ini -file:

```
[pytest]
asyncio_mode=auto
python_files = *_features.py
python_classes = *Features
python_functions = should_*
```

Create two preliminary fixtures:

- A fixture providing a db connection. This connection uses in-memory SQLite instead of PostgreSQL
- A fixture providing our REST client. This fixture uses the db connection fixture.

Acceptance criteria: backend has a pytest test set and aforementioned fixtures. Root route (hello world-return) is tested.

### PR DESCRIPTION
**Pull request by EddieTheCubeHead**
Combined the tickets as SQLAlchemy was dead code without tests.

- Added sqlalchemy connection manager with configurable connection string in src/database/database_connection
- Added pytest.ini to configure pytest with initial configuration values
- Added test/conftest to house pytest fixtures
  - Added fixtures for database client and test server
- Added framework_features -feature test file to host tests that validate framework functionality
  - Added tests for FastAPI root route and SQLAlchemy client basic functionality in here
- Added documentation for running tests locally
- Added test_requirements.txt to separate functionally required packages from packages required to run test set

Closing this also closes #8

### CODE REVIEW

File: server/src/api/routes.py [hunk 1]
```diff
@@ -3,4 +3,4 @@
 
 @application.get("/")
 async def root():
-    return {"message": "Hello Worlds"}
+    return {"message": "Hello World!"}
```
Comment (EddieTheCubeHead) [inline server/src/api/routes.py:6]: I had accidentally edited this while validating auto-reload works. Edited back

File: server/pytest.ini [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[pytest]
+python_files = *_features.py
+python_classes = *Features
+python_functions = should_*
+pythonpath = src
```
Comment (EddieTheCubeHead) [inline server/pytest.ini:5]: This enables pytest to be run outside of src without adding src to pythonpath

File: server/pytest.ini [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[pytest]
+python_files = *_features.py
+python_classes = *Features
+python_functions = should_*
```
Comment (EddieTheCubeHead) [inline server/pytest.ini:4]: These three lines configure how pytest discovers tests.

- Look for files that end with `_features.py`
- Look for classes in those files that end with `_features.py`
- Look for functions inside those files and classes that start with `should_`

**Review by KeManen — APPROVED** [review]

**Review by BadhonRahman01 — APPROVED** [review]
looks good!

**Pull request merged**